### DISCUSSION TITLE
Cannot run docker in docker

### DISCUSSION THREAD

**Discussion by GCSBOSS** [apocas/dockerode#740] [Q&A]
I'm trying to run DinD to emulate a swarm for tests.

```js
const node1 = await docker.createContainer({
    Image: 'docker:dind',
    Tty: false,
    Init: true,
    Privileged: true
});
await node1.start();

const exec = await node1.exec({
    Cmd: [ 'docker', 'swarm', 'init' ],
    // AttachStdin: true,
    AttachStdout: true,
    AttachStderr: true,
    Privileged: true
});

const stream = await exec.start({ stdout: true/*hijack: true, stdin: true*/ });

docker.modem.demuxStream(stream, process.stdout, process.stderr);
```

When I run this it spits:

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`
```

Reading [this SO question](https://stackoverflow.com/questions/49390573/docker-dind-cannot-connect-to-the-docker-daemon) I'm not sure if somehow my `Privileged` flag is not working or what else may be the issue.

**Comment by SukeshRondla** [comment]
Host Docker Socket Mounting: When you run DinD, you need to mount the Docker socket from the host system into the DinD container so that the DinD container can communicate with the host's Docker daemon. This is typically done using a volume mount like -v /var/run/docker.sock:/var/run/docker.sock.

Modify your code to include this volume mount:
javascript
Copy code
```
const node1 = await docker.createContainer({
    Image: 'docker:dind',
    Tty: false,
    Init: true,
    Privileged: true,
    HostConfig: {
        Binds: ['/var/run/docker.sock:/var/run/docker.sock']
    }
});
```

**Reply by GCSBOSS** [reply]
Still having the same issue after adding the mount. 

I wonder why whould the docker image need to connect with the host daemon?
